src/app/text-marking/text-marking.ts

```diff
@@ -99,10 +111,132 @@ export function markText(
             } else {
                 throw Error('Unexpected node type!');
             }
+
+        }
+
+    }
+
+    isLastCall(textMarkings.length, typoElements.length)
+
+}
+
+
+/**
+ * Updates the state of text markings and applies animations to the HTML element based on changes detected
+ * after invoking the `markText` function.
+ *
+ * @param {HTMLElement} newNode - The newly created or modified HTML element.
+ * @param {string} newTextContent - The text content of the HTML element.
+ *
+ * This function adds the new text content to the `highlightedMarkingWords` array.
+ * It then checks the previous state (`lastHighlightedMarkingWords`) and determines
+ * whether to apply animations to the nodes based on the changes detected.
+ * The logic considers scenarios where nodes are added, removed, or modified.
+ */
+function makeTextMarkings(newNode: HTMLElement, newTextContent: string){
+    highlightedMarkingWords.push(newTextContent)
+
+    // Check if there are any previous markings
+    if(lastHighlightedMarkingWords.length === 0){
+        // If no previous markings, apply animation to the new node
+        newNode.style.animation = 'fillBackground 500ms linear forwards';
+    }else{
+        // Check various scenarios based on changes in markings
+        if(lastHighlightedMarkingWords.length < markingCount){
+            // If fewer markings than expected, apply animation to the new node
+            newNode.style.animation = 'fillBackground 500ms linear forwards';
+        }else if(lastHighlightedMarkingWords[markingCount] === newTextContent){
+            // If the current marking is the same, remove animation
+            newNode.style.animation = '';
+        }else if(lastHighlightedMarkingWords[markingCount] !== newTextContent){
+            // Handle scenarios for additions, removals, and modifications
+            if(lastHighlightedMarkingWords[markingCount + 1] === newTextContent){
+                newNode.style.animation = '';
+                // Remove the item if it was changed back to the previous content
+                removeItemAtIndex(markingCount)
+            }else if(markingCount === 0 && lastHighlightedMarkingWords[markingCount + 1] !== newTextContent){
+                // If a new first node was added, insert it and apply animation
+                addItemAtIndex(0, newTextContent)
+                newNode.style.animation = 'fillBackground 500ms linear forwards';
+            }else if(lastHighlightedMarkingWords[markingCount - 1] !== newTextContent || lastHighlightedMarkingWords[markingCount + 1] !== newTextContent){
+                // If the node was modified or inserted somewhere in the middle, apply animation
+                addItemAtIndex(markingCount, newTextContent)
+                newNode.style.animation = 'fillBackground 500ms linear forwards';
+            }else{
+                newNode.style.animation = 'fillBackground 500ms linear forwards';
```

Comment:
Consider using classes instead of assigning CSS as strings in JS. It keeps the CSS code in CSS files.

If you still want to keep it like this, you can create a const for `'fillBackground 500ms linear forwards'` since if I'm not mistaken it is repeated many times.